core/src/main/scala/com/yahoo/maha/core/dimension/Dimension.scala

```diff
@@ -645,8 +646,9 @@ object Dimension {
                     , viewBaseTable : Option[String] = None
                     , isDerivedDimension: Boolean = false
                     , underlyingTableName : Option[String] = None
+                    , secondaryDimLevel: Option[Int] = Some(1)
                     ) : DimensionBuilder = {
-    val baseFact = new DimTable(name, 9999, engine, dimLevel, schemas, columns, None, schemaColMap, annotations, ddlAnnotation, isDerivedDimension, viewBaseTable, maxDaysLookBack, underlyingTableName)
```

Comment:
add check here that secondaryDimLevel should be defined. as you are accessing it secondaryDimLevel.get in the RequestModel